ls1intum/Artemis
# Bugfix/questions answers/edit answers

### Checklist

### Motivation and Context



(BUG1) TypeError is thrown when updating an answer after creation.

(BUG2) When editing an answer choosing to edit another answer made the new one disappear.

(ENHANCEMENT) The cursor doesn't indicate that clicking the _show more / less answers_ option is possible

### Description

(BUG1) converted date with moment before checking to prevent TypeError

(BUG2) check all combinations while toggling answerMode to prevent unwanted behaviour

(ENHANCEMENT) the cursor is now a pointer while hovering over the _show more / less answers_ option to indicate it is clickable

### Steps for Testing


1. Log in to Artemis (and activate answers and questions for your test course if not already)
2. Navigate to an Exercise / Lecture
3. Create multiple questions
4. Create multiple answers for each question
5. approve an answers of your choice
6. (ENHANCEMENT) hover over _show less / more answers_
7. (BUG1) Update the text of an answer and check console for errors
8. (BUG2) Click on the edit button of an answer to activate the edit mode
9. (BUG2) Click on it again
10. (BUG2) Click on the edit button of another answer 

### Screenshots


![bugfix-edit](https://user-images.githubusercontent.com/51987021/79051033-2c317500-7c2e-11ea-8879-1997ab481dba.gif)
